pkg/plugin/aggregation/aggregator.go

```diff
@@ -217,5 +225,20 @@ func (a *Aggregator) handleResult(result *plugin.Result) error {
 	io.Copy(f, result.Body)
 	glog.Infof("wrote results to %v\n", resultsFile)
 
+	// If it's a tarball, extract it
+	if extension == ".tar.gz" {
+		f.Close()
+		err = tarx.Extract(resultsFile, path.Join(a.OutputDir, result.Path()), &tarx.ExtractOptions{})
+		if err != nil {
+			glog.Errorf("Could not extract tar file %v: %v", resultsFile, err)
+			return err
```

Comment:
Good point, currently errors are channeled from `(*Driver).Monitor()` as `Result` objects and get stored the same way, but nothing's specifying that their extension should be json.  I'll add `Extension: ".json"` to the MakeErrorResult function [here](https://github.com/heptio/sonobuoy/blob/master/pkg/plugin/driver/utils/utils.go#L77), and make sure we have tests for this.